client/web/web.go

```diff
@@ -530,13 +539,42 @@ func (s *Server) tailscaleUp(ctx context.Context, st *ipnstate.Status, postData
 	}
 }
 
-// csrfKey creates a new random csrf token.
-// If an error surfaces during key creation,
-// the error is logged and the active process terminated.
-func csrfKey() []byte {
+// csrfKey returns a key that can be used for CSRF protection.
+// If an error occurs during key creation, the error is logged and the active process terminated.
+// If the server is running in CGI mode, the key is cached to disk and reused between requests.
+// If an error occurs during key storage, the error is logged and the active process terminated.
+func csrfKey(cgiMode bool) []byte {
```

Comment:
Maybe switch this to `func (s *Server) csrfKey()`? Then could just check `s.cgiMode`

Feels like it can be a receiver now that the key logic is a little more specific to the web client setup.

Reply:
done